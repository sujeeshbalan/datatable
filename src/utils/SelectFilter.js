import React from 'react';
import PropTypes from 'prop-types';

// Import React-Bootstrap
import { FormGroup, FormControl, InputGroup, Button } from 'react-bootstrap';

import FontAwesome from './FontAwesome';

class SelectFilter extends React.Component {
  onInputChange = (e) => {
    this.props.onChangeFilter(e.target.value);
  }

  onClearFilter = (e) => {
    e.preventDefault();

    this.props.onChangeFilter('');
  }

  render() {
    return (
      <FormGroup controlId={`select-filter-${this.props.keyName}`}>
        <InputGroup>
          <FormControl
            type="text"
            value={this.props.filterText}
            placeholder="Enter text"
            onChange={this.onInputChange}
          />
          <InputGroup.Button>
            <Button onClick={this.onClearFilter}>
              <FontAwesome icon="times" additionalClass="fa-fw" />
            </Button>
          </InputGroup.Button>
        </InputGroup>
      </FormGroup>
    );
  }
}

SelectFilter.propTypes = {
  filterText: PropTypes.string.isRequired,
  keyName: PropTypes.string.isRequired,
  onChangeFilter: PropTypes.func.isRequired,
};

export default SelectFilter;
