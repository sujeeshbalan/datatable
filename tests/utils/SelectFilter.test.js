import {
  FormGroup,
  FormControl,
  InputGroup,
  Button,
} from 'react-bootstrap';

import SelectFilter from '../../src/utils/SelectFilter';
import FontAwesome from '../../src/utils/FontAwesome';

function setup() {
  const props = {
    filterText: 'stub filter',
    keyName: 'selectFilterID',
    onChangeFilter: jest.fn(),
  };

  const enzymeWrapper = shallow(
    <SelectFilter {...props} />
  );

  return {
    props,
    enzymeWrapper
  };
}

describe('SelectFilter component (js/component/SelectFilter)', () => {
  it('should match normal snapshot', () => {
    const { props, enzymeWrapper } = setup();

    expect(enzymeWrapper).toMatchSnapshot();
  });

  it('should render self and subcomponents', () => {
    const { props, enzymeWrapper } = setup();

    expect(enzymeWrapper.instance().props.filterText).toBe('stub filter');
    expect(enzymeWrapper.instance().props.keyName).toBe('selectFilterID');

    expect(enzymeWrapper.find(FormGroup).props().controlId).toBe('select-filter-selectFilterID');
    expect(Object.keys(enzymeWrapper.find(FormGroup).props()).length).toBe(3);
    expect(enzymeWrapper.find(FormGroup).length).toBe(1);

    expect(enzymeWrapper.find(InputGroup).length).toBe(1);

    expect(enzymeWrapper.find(FormControl).props().type).toBe('text');
    expect(enzymeWrapper.find(FormControl).props().value).toBe('stub filter');
    expect(enzymeWrapper.find(FormControl).props().placeholder).toBe('Enter text');
    expect(typeof enzymeWrapper.find(FormControl).props().onChange).toBe('function');
    expect(enzymeWrapper.find(FormControl).length).toBe(1);

    expect(enzymeWrapper.find(InputGroup.Button).length).toBe(1);

    expect(typeof enzymeWrapper.find(Button).props().onClick).toBe('function');
    expect(enzymeWrapper.find(Button).length).toBe(1);

    expect(enzymeWrapper.find(FontAwesome).props().icon).toBe('times');
    expect(enzymeWrapper.find(FontAwesome).props().additionalClass).toBe('fa-fw');
    expect(enzymeWrapper.find(FontAwesome).length).toBe(1);
  });

  it('should trigger onClearFilter', () => {
    const { props, enzymeWrapper } = setup();

    const value = '';
    enzymeWrapper.find(Button).simulate('click', {
      preventDefault: jest.fn(),
    });

    expect(props.onChangeFilter).toBeCalledWith(value);
    expect(props.onChangeFilter.mock.calls.length).toBe(1);
  });

  it('should trigger onInputChange with a certain value', () => {
    const { props, enzymeWrapper } = setup();

    const value = 'asd';
    enzymeWrapper.find(FormControl).simulate('change', {
      target: {value},
    });

    expect(props.onChangeFilter).toBeCalledWith(value);
    expect(props.onChangeFilter.mock.calls.length).toBe(1);
  });
});
