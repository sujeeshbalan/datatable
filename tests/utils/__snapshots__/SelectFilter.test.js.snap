// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFilter component (js/component/SelectFilter) should match normal snapshot 1`] = `
<FormGroup
  bsClass="form-group"
  controlId="select-filter-selectFilterID"
>
  <InputGroup
    bsClass="input-group"
  >
    <FormControl
      bsClass="form-control"
      componentClass="input"
      onChange={[Function]}
      placeholder="Enter text"
      type="text"
      value="stub filter"
    />
    <InputGroupButton
      bsClass="input-group-btn"
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        onClick={[Function]}
      >
        <FontAwesome
          additionalClass="fa-fw"
          icon="times"
        />
      </Button>
    </InputGroupButton>
  </InputGroup>
</FormGroup>
`;
